Src/Support/IntegrationTests/PerCallAuthTests.cs

```diff
@@ -61,6 +64,151 @@ public void OverrideServiceCredentialWithPerCallCredential()
             Assert.NotNull(buckets.Items);
         }
 
+        private class RequestInterceptorCredentialWrapper : IConfigurableHttpClientInitializer, IHttpExecuteInterceptor
+        {
+            private AccessTokenCredential _credential;
+            public HttpRequestMessage Request { get; private set; }
+
+            public RequestInterceptorCredentialWrapper(string accessToken, IAccessMethod accessMethod) =>
+                _credential = new AccessTokenCredential(accessToken, accessMethod);
+
+            public void Initialize(ConfigurableHttpClient httpClient)
+            {
+                _credential.Initialize(httpClient);
+                httpClient.MessageHandler.AddExecuteInterceptor(this);
+            }
+
+            public Task InterceptAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+            {
+                Request = request;
+                return Task.FromResult(true);
+            }
+        }
+
+        [Fact]
+        public void OverridesServiceCredentialWithPerCallCredential_QueryString()
+        {
+            var serviceCredential = new RequestInterceptorCredentialWrapper("SERVICE_CREDENTIAL_TOKEN", new BearerToken.QueryParameterAccessMethod());
+            var callCredential = new AccessTokenCredential("CALL_CREDENTIAL_TOKEN", new BearerToken.QueryParameterAccessMethod());
+
+            // Set one credential in the client
```

Comment:
This ~10 line block is repeated a lot. I think it would make sense to extract that to a helper method accepting a service credential and a call credential, and returning an HttpRequest.